packages/headless/src/utils/validate-payload.ts

```diff
@@ -24,6 +25,11 @@ export const requiredEmptyAllowedString = new StringValue({
   emptyAllowed: true,
 });
 
+export const nonEmptyArray = new ArrayValue({
```

Comment:
Given this accepts empty arrays as you said maybe we should rename it to
`nonEmptyStringArray`